avocado/core/job.py

```diff
@@ -285,6 +286,11 @@ def _make_test_suite(self, urls=None):
                      Optionally, a list of tests (each test a string).
         :returns: a test suite (a list of test factories)
         """
+        suite = [(None, {})]
```

Comment:
Well, I giving this a thought now... I'm not sure the loader should be engaged in sort out when a test is remote or not. We have the job, which is local, the RemoteRunner, which runs the test remotely and the loader, which IMO should not be called by the local job when the test is remote. The job is the entity aware about the "remote" principle of the runner.
Anyway, I agree faking the suite is not the best approach. We should instead accept `None` as a valid value for suite and don't try to handle the suite when it's `None`. loader will then only be involved when the test is local. @clebergnu how does this sound for you?